### ISSUE TITLE
Question: Power Calculation?

### ISSUE THREAD

**Issue by tdurack** [Sublimis/UrbanBiker#82]
Does the individual component weight make any difference to the power calc? Is it important to break-out bike, wheels, bag weight separately for an accurate calculation? Or can I simply set bike weight to include bike+wheels+bags?

**Comment by Sublimis** [comment]
Only wheels weight has special status. It's important to fill in proper wheels weight so rotational inertia can be used for more accurate power calculations.

How you break-out the total weight (everything, including wheels) in other boxes is irrelevant for power calculation. You can simply put the whole weight into the "Vehicle" box if you wish. Multiple boxes are given just so it's easier for you to e.g. include/exclude bags in total weight when you mount them or take them down. Hope this helps!

**Comment by tdurack** [comment]
Got it - thanks!

**Comment by tdurack** [comment]
Wheel weight for rotational inertia: should this be mainly tire+tube(less)+rim+spokes? Should hub+rotor+cassette+skewer/axle be included or excluded?

Not trying to be crazy, would rather just weigh the two wheels and put that in the appropriate box, but I also like accurate power calcs...

**Comment by Sublimis** [comment]
The app pretends the whole weight of a wheel (including that of all its parts, cassette, hub etc.) resides on the circumference. So yes, the hub, rotor, spokes and cassette disturb this idealized picture a bit. But you should not simply subtract their weight from the total wheel weight; they are also rotating and contributing to rotational inertia. Instead, if you are to be really pedantic about this, you should subtract only part of hub+rotor+spokes+cassette weight. But which part? No one can tell, but an estimate can be given.

If you look [here](https://en.wikipedia.org/wiki/List_of_moments_of_inertia#Moments_of_inertia) you'll see the moment of inertia of a unit circular loop is 1/2 and that of a solid disk is half that, 1/4. Bicycle wheel is somewhere between a circular loop and a solid disk. If we wanted to make an estimate, we would say subtract 50% of hub+rotor+spokes+cassette from the total wheel weight. Even better, make measurements, if possible. 😉 

**To sum up**: Bike (vehicle) weight should include everything. Simply weigh the whole bike. Wheels weight should include weight of both wheels, but you can then _slightly_ reduce this number to adjust for the idealization mentioned above.

**Comment by tdurack** [comment]
I re-read the manual and realized this was explained. Thanks for your patience!

**Comment by tdurack** [comment]
One more related question: Is it possible to edit power coefficients in a recorded history track and have that reflected in the recorded power values?

(Reason: I am tuning coefficients and want to update already recorded tracks)

**Comment by Sublimis** [comment]
Sorry, no. While in theory it could be possible to implement some sort of "recalculate power" procedure, which would consider SVG graphs as well as GPX file, it's questionable how accurate this would be in post-processing stage. The data in SVG and GPX files has already been through digital filters, and the raw original data is long gone. At this time it seems like too much work, for uncertain results...

**Comment by tdurack** [comment]
Makes sense, thanks
